docs/expected-to-fail-tests.adoc

```diff
@@ -45,11 +45,9 @@ include::{demo}[tag=expected_to_fail_message]
 ----
 
 In some cases it may be desirable to only have the test be aborted on certain exceptions.
-In the following example a test case for `productionFeature()` has been implemented. The test
-has been fully implemented, but the production code has not been implemented yet and throws an
-`UnsupportedOperationException` to indicate this. Once the production code has been implemented
-the test method will fail if the code were to return the wrong result and fail due to a failing
-assertion.
+In the following example a test case for `productionFeature()` has been implemented.
+The test has been fully implemented, but the production code has not been implemented yet and throws an `UnsupportedOperationException` to indicate this.
+Once the production code has been implemented the test method will fail if the code were to return the wrong result and fail due to a failing assertion.
```

Comment:
Wait, is this correct? I thought it would always fail because the expectation is not fulfilled so it doesn't matter what's in the test.

Reply:
Maybe you can propose a better way to phrase this. By "Once the production code has been implemented" I mean that `productionFeature()` is now no longer throwing an `UnsupportedOperationException`, but instead maybe the wrong answer (e.g. `11`). Without `onExceptions` the test would continue to be skipped. With `onExceptions` the failing assertion will now cause the test to fail.

Perhaps it would make it clearer if I spell it out like that.

Reply:
```
Once the feature has been implemented `@ExpectedToFail` will fail the test as it no longer throws `UnsupportedOperationException`. If you don't use `onExceptions` in `@ExpectedToFail` you risk "masking" your implementation being faulty - which `@ExpectedToFail` would mark as passing.
```

What do you think?

I would also put a warning here to NOT use `onExceptions` for assertion - you can look at Nicolai's comment on the linked issue.

Reply:
I've rephrased the documentation yet yet again (and added a subsection header). I hope this will clarify things.

Please note that I am of course also happy with you changing the documentation to whatever you like. I am obviously not a technical writer 😄

Reply:
@Michael1993 given your approval on December 2, 2023, I mark this as resolved.